### PR TITLE
[WFCORE-5843] [primary/secondary] Find and replace occurrences in the…

### PR DESCRIPTION
**Pull request by moulalis**
Issue: https://issues.redhat.com/browse/WFCORE-5843

### CODE REVIEW

File: elytron/src/test/java/org/wildfly/extension/elytron/CredentialStoreUtility.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ public CredentialStoreUtility(String credentialStoreFileName, String storePasswo
             LOGGER.error(t);
             throw new RuntimeException(t);
         }
-        LOGGER.debugf("Credential Store created [%s] with master password \"%s\"", credentialStoreFileName, storePassword);
+        LOGGER.debugf("Credential Store created [%s] with primary password \"%s\"", credentialStoreFileName, storePassword);
```
Comment (yersan) [inline elytron/src/test/java/org/wildfly/extension/elytron/CredentialStoreUtility.java:98]: So far the concept of primary/secondary is just for "Domain Model", see WFLY-14053. +1 for discovering this, but this one would need to be treated differently. 

@fjuma Do you have any suggestions for this word?

File: host-controller/src/main/java/org/jboss/as/host/controller/Main.java [hunk 2]
```diff
@@ -367,7 +367,7 @@ public static HostControllerEnvironmentWrapper determineEnvironment(String[] arg
                     if (initialHostConfig == null) {
                         return new HostControllerEnvironmentWrapper(HostControllerEnvironmentWrapper.HostControllerEnvironmentStatus.ERROR);
                     }
-                } else if (arg.startsWith(CommandLineConstants.MASTER_ADDRESS)) {
+                } else if (arg.startsWith(CommandLineConstants.PRIMARY_ADDRESS)) {
```
Comment (yersan) [inline host-controller/src/main/java/org/jboss/as/host/controller/Main.java:370]: Please, try to focus the changes on this PR to only messages of the loggers, that will avoid potential conflicts and keep the changes related to its issue.

**Review by yersan — CHANGES_REQUESTED** [review]
Hi @moulalis,

I added some feedback, in general, 
- try to limit the changes to the Jira intention
- I still miss loggers that need a review (DomainManagementLogger, DomainControllerLogger, HostControllerLogger, ServerLogger.java (we can use main instead of master))

In case it helps, I'm using the following command for logger discovery:

git grep "master" | grep "/logging/" | grep "@Message"
git grep "slave" | grep "/logging/" | grep "@Message"

File: elytron/src/test/java/org/wildfly/extension/elytron/CredentialStoreUtility.java (hunk 1 above)
Comment (yersan) [reply elytron/src/test/java/org/wildfly/extension/elytron/CredentialStoreUtility.java:98]: perfect, thanks @fjuma

File: process-controller/src/main/java/org/jboss/as/process/CommandLineConstants.java [hunk 3]
```diff
@@ -120,8 +120,8 @@
 
     public static final String ADMIN_ONLY = "--admin-only";
 
-    public static final String MASTER_ADDRESS = "--master-address";
-    public static final String MASTER_PORT = "--master-port";
+    public static final String PRIMARY_ADDRESS = "--primary-address";
+    public static final String PRIMARY_PORT = "--primary-port";
```
[the remark this answers is not in the document]
Comment (moulalis) [reply process-controller/src/main/java/org/jboss/as/process/CommandLineConstants.java]: This is removed.

File: elytron/src/test/java/org/wildfly/extension/elytron/CredentialStoreUtility.java (hunk 1 above)
Comment (moulalis) [reply elytron/src/test/java/org/wildfly/extension/elytron/CredentialStoreUtility.java]: master is dropped from logger

File: host-controller/src/main/java/org/jboss/as/host/controller/Main.java (hunk 2 above)
Comment (moulalis) [reply host-controller/src/main/java/org/jboss/as/host/controller/Main.java]: @yersan PR is updated. Thanks.

File: host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java [hunk 4]
```diff
@@ -681,13 +681,13 @@
...
      * Creates an exception indicating the need to call the method, represented by the {@code methodName} parameter,
-     * before checking the slave status.
+     * before checking the secondary status.
      *
      * @param methodName the name of the method to invoke.
      *
      * @return an {@link IllegalStateException} for the error.
      */
-    @Message(id = 67, value = "Must call %s before checking for slave status")
+    @Message(id = 67, value = "Must call %s before checking for secondary status")
```
Comment (yersan) [inline host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java:690]: s/for secondary status/for secondary host controller status

File: host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java [hunk 5]
```diff
@@ -1107,18 +1107,18 @@
 //    @Message(id=139, value="Invalid value for %s. Must only contain all of the existing discovery options")
 //    OperationFailedException invalidDiscoveryOptionsOrdering(String name);
 
-    @Message(id=140, value="Can't execute transactional operation '%s' from slave controller")
+    @Message(id=140, value="Can't execute transactional operation '%s' from secondary controller")
```
Comment (yersan) [inline host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java:1110]: s/from secondary controller/from the secondary controller

File: host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java [hunk 6]
```diff
@@ -1107,18 +1107,18 @@
...
 //    OperationFailedException noResourceFor(PathAddress address);
 
     @LogMessage(level = Level.ERROR)
-    @Message(id=142, value = "Failed to apply domain-wide configuration from master host controller")
+    @Message(id=142, value = "Failed to apply domain-wide configuration from primary host controller")
     void failedToApplyDomainConfig(@Cause Exception e);
 
     @LogMessage(level = Level.ERROR)
-    @Message(id=143, value = "Failed to apply domain-wide configuration from master host controller. " +
+    @Message(id=143, value = "Failed to apply domain-wide configuration from primary host controller. " +
```
Comment (yersan) [inline host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java:1121]: s/from primary host controller/from the primary host controller

File: host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java [hunk 7]
```diff
@@ -1142,13 +1142,13 @@ void noAccessControlConfigurationAvailable(RunningMode currentRunningMode, Strin
...
     /**
-     * Logs a warning message indicating that the slave host controller could not
+     * Logs a warning message indicating that the secondary host controller could not
      * discover the remote domain controller using the given {@link org.jboss.as.host.controller.discovery.DiscoveryOption}.
      *
      * @param e the cause of the error.
      */
     @LogMessage(level = Level.WARN)
-    @Message(id = 146, value = "Could not discover master using discovery option %s. Error was: %s")
+    @Message(id = 146, value = "Could not discover primary using discovery option %s. Error was: %s")
```
Comment (yersan) [inline host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java:1151]: s/discover primary using/discover the primary host controller using

File: host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java [hunk 8]
```diff
@@ -1160,20 +1160,20 @@ void noAccessControlConfigurationAvailable(RunningMode currentRunningMode, Strin
...
     /**
-     * Logs a message indicating that the slave host controller connected with the master HC.
+     * Logs a message indicating that the secondary host controller connected with the primary HC.
      *
-     * @param uri the URI at which the master was reached
+     * @param uri the URI at which the primary was reached
      */
     @LogMessage(level = INFO)
-    @Message(id = 148, value = "Connected to master host controller at %s")
+    @Message(id = 148, value = "Connected to primary host controller at %s")
```
Comment (yersan) [inline host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java:1168]: s/Connected to primary host/Connected to the primary host

File: host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java [hunk 9]
```diff
@@ -1160,20 +1160,20 @@ void noAccessControlConfigurationAvailable(RunningMode currentRunningMode, Strin
...
     void connectedToMaster(URI uri);
 
     @LogMessage(level = INFO)
     @Message(id = 149, value = "Option %s was set; obtaining domain-wide configuration from %s")
     void usingCachedDC(String configOption, String cachedXmlFile);
 
     @LogMessage(level = INFO)
-    @Message(id = 150, value = "Trying to reconnect to master host controller.")
+    @Message(id = 150, value = "Trying to reconnect to primary host controller.")
```
Comment (yersan) [inline host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java:1176]: s/to primary host/to the primary host

File: process-controller/src/main/java/org/jboss/as/process/logging/ProcessLogger.java [hunk 10]
```diff
@@ -433,23 +433,23 @@
...
+    @Message(id = Message.NONE, value = "Set the host controller's running type to ADMIN_ONLY causing it to open administrative interfaces and accept management requests but not start servers or, if this host controller is the primary for the domain, accept incoming connections from secondary host controllers.")
     String argAdminOnly();
 
     /**
      * Instructions for the {@link CommandLineConstants#MASTER_ADDRESS} command line argument.
      *
      * @return the message.
      */
-    @Message(id = Message.NONE, value = "Set system property jboss.domain.master.address to the given value. In a default slave Host Controller config, this is used to configure the address of the master Host Controller.")
+    @Message(id = Message.NONE, value = "Set system property jboss.domain.primary.address to the given value. In a default secondary Host Controller config, this is used to configure the address of the primary Host Controller.")
```
Comment (yersan) [inline process-controller/src/main/java/org/jboss/as/process/logging/ProcessLogger.java:444]: Don't change the property name, here we are changing only messages of the loggers, properties are tracked on a different Jira.

File: process-controller/src/main/java/org/jboss/as/process/logging/ProcessLogger.java [hunk 11]
```diff
@@ -433,23 +433,23 @@
...
     String argMasterAddress();
 
     /**
      * Instructions for the {@link CommandLineConstants#MASTER_PORT} command line argument.
      *
      * @return the message.
      */
-    @Message(id = Message.NONE, value = "Set system property jboss.domain.master.port to the given value. In a default slave Host Controller config, this is used to configure the port used for native management communication by the master Host Controller.")
+    @Message(id = Message.NONE, value = "Set system property jboss.domain.primary.port to the given value. In a default secondary Host Controller config, this is used to configure the port used for native management communication by the primary Host Controller.")
```
Comment (yersan) [inline process-controller/src/main/java/org/jboss/as/process/logging/ProcessLogger.java:452]: Don't change the property name, here we are changing only messages of the loggers, properties are tracked on a different Jira.

**Review by yersan — CHANGES_REQUESTED** [review]
Hello @moulalis. I've added some reviews. Notice that not always a direct replacement works fine. `master` and `slaves` are nouns and the meaning of them is well known (master as domain controller or primary host controller, and slave as the other host controllers), but primary/secondary is quite new so sometimes we need to complement them adding more information. Also, take into account that primary/secondary are not nouns.

File: server/src/main/java/org/jboss/as/server/logging/ServerLogger.java [hunk 12]
```diff
@@ -592,7 +592,7 @@
      *
      * @return the message
      */
-    @Message(id = Message.NONE, value = "The git branch to use to get the server configuration. Default is 'master'")
+    @Message(id = Message.NONE, value = "The git branch to use to get the server configuration. Default is 'main'")
```
Comment (bstansberry) [inline server/src/main/java/org/jboss/as/server/logging/ServerLogger.java:595]: Please revert this change. The default is 'master'.  This describes the name of a branch in a user controlled git repository, not the name of a branch in any repo under the control of the WildFly project. Changing the default is a breaking change and not something we should mix with the rest of the things in this PR.

**Review by bstansberry — COMMENTED** [review]
@moulalis @yersan 

I made one comment below, but I'll have a proper look at this later this week. I want to see if I can get some feedback from a couple folks on what we're doing before I do a full review.

File: server/src/main/java/org/jboss/as/server/logging/ServerLogger.java (hunk 12 above)
Comment (yersan) [reply server/src/main/java/org/jboss/as/server/logging/ServerLogger.java]: You're right, thanks, I passed it over.

**Review by yersan — CHANGES_REQUESTED** [review]
Hi @moulalis , you pushed up the PR but there were still some pending points to fix. I've reviewed it again and found some additional ones. Would you mind checking it again?

**Review by yersan — CHANGES_REQUESTED** [review]
Hi @moulalis , you pushed up the PR but there were still some pending points to fix. Would you mind checking it again? Please, ping me when the PR is ready to review, thanks!

File: host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java [hunk 13]
```diff
@@ -320,12 +320,12 @@
...
 
     /**
-     * Logs an informational message indicating the host has been unregistered as a remote slave.
+     * Logs an informational message indicating the host has been unregistered as a remote secondary.
      *
      * @param host the host.
      */
     @LogMessage(level = INFO)
-    @Message(id = 26, value = "Unregistered remote slave host \"%s\"")
+    @Message(id = 26, value = "Unregistered remote secondary host \"%s\"")
```
[the remark this answers is not in the document]
Comment (yersan) [reply host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java:328]: Let's resolve this, since we are using a name for the host, it looks like the phase is dramatically correct.

File: host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java [hunk 14]
```diff
@@ -250,13 +250,13 @@
...
     /**
-     * Logs an informational message indicating the host has been registered as a remote slave.
+     * Logs an informational message indicating the host has been registered as a remote secondary.
      *
      * @param hostName the host name
      * @param productName the product name
      */
     @LogMessage(level = INFO)
-    @Message(id = 19, value = "Registered remote slave host \"%s\", %s")
+    @Message(id = 19, value = "Registered remote secondary host \"%s\", %s")
```
[the remark this answers is not in the document]
Comment (yersan) [reply host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java:259]: Let's resolve this, since we are using a name for the host, it looks like the phase is dramatically correct.

[the remark this answers is not in the document]
Comment (moulalis) [reply host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java]: PR updated.

**Review by yersan — APPROVED** [review]
LGTM, but I would like to see a second review. 

@bstansberry Could you take a look at this one? All comments were already resolved, there is nothing pending.

Just take into account that jboss.domain.master.port/jboss.domain.master.address changes are tracked on a different Jira.

**Review by bstansberry — CHANGES_REQUESTED** [review]
@moulalis Sorry I reviewed this long ago but apparently never submitted it.

File: host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java [hunk 15]
```diff
@@ -250,13 +250,13 @@
     void reconnectingServer(String serverName);
 
     /**
-     * Logs an informational message indicating the host has been registered as a remote slave.
+     * Logs an informational message indicating the host has been registered as a remote secondary host.
```
[the remark this answers is not in the document]
Comment (bstansberry) [reply host-controller/src/main/java/org/jboss/as/host/controller/logging/HostControllerLogger.java:253]: Never mind; host is ok.

File: host-controller/src/main/java/org/jboss/as/domain/controller/logging/DomainControllerLogger.java [hunk 16]
```diff
@@ -475,11 +475,11 @@
     String noDeploymentContentWithHash(String hash);
 
     /**
-     * A message indicating a slave domain controller cannot accept deployment content uploads.
+     * A message indicating a secondary domain controller cannot accept deployment content uploads.
```
[the remark this answers is not in the document]
Comment (moulalis) [reply host-controller/src/main/java/org/jboss/as/domain/controller/logging/DomainControllerLogger.java]: PR Updated please check and approve it. Thanks.

**Comment by yersan** [comment]
@bstansberry I'm going to merge this now to unlock other PRs. I've checked your changes were already addressed, so we should be fine.
